src/components/media/Media.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+import * as React from 'react';
+import classnames from 'classnames';
+import MediaImg from './MediaImg';
+import MediaBody from './MediaBody';
+import MediaMenu from './MediaMenu';
+import './Media.scss';
+
+type Props = {
+    /** Component to use as outermost element, e.g., 'li' */
+    as: React.ElementType,
+    /** Child elements */
+    children: React.ChildrenArray<React.Element<typeof MediaImg | typeof MediaBody | typeof MediaMenu>>,
+    /** Additional class names */
+    className?: string,
+};
+
+function Media({ as: Wrapper, children, className, ...rest }: Props) {
+    return (
+        <Wrapper className={classnames('bdl-Media', className)} {...rest}>
+            {children}
+        </Wrapper>
+    );
+}
+
+Media.defaultProps = {
+    as: 'div',
```

Comment:
Can we add semantic meaning to the markup? Since the intended use is for an image with related content, would a `figure/figcaption` pairing make sense?

Reply:
Does that make sense if it's a text-only avatar?

Reply:
Perhaps the naming is confusing, as `MediaImage` also doesn't fit text content very well.

Reply:
In the Bootstrap implementation, the first child is just the media content, it doesn't have a wrapper. We could do that, but I'm not a huge fan because you have to use .cloneElement to add vertical alignment options, etc.

Some ideas:

- `Media.Content` 
- `Media.Left`
- `Media.Object`

Prior Art:

- https://react-bootstrap.github.io/layout/media/
- https://foundation.zurb.com/sites/docs/media-object.html
- https://bulma.io/documentation/layout/media-object/

Reply:
MDN says other flow content is inside figure seems OK so maybe we _should_ default to that. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figure#Usage_notes

Reply:
Going with `<Media.Figure />` and defaulting to `figure` within